Schemathesis on group members

```
  grpc.aio._call.AioRpcError: <AioRpcError of RPC that terminated with:
  	status = StatusCode.INVALID_ARGUMENT
  	details = "invalid ReadRelationshipsRequest.RelationshipFilter: embedded message failed validation | caused by: invalid RelationshipFilter.OptionalSubjectFilter: embedded message failed validation | caused by: invalid SubjectFilter.OptionalSubjectId: value does not match regex pattern "^(([a-zA-Z0-9/_|\\-=+]{1,})|\\*)?$""
  	debug_error_string = "UNKNOWN:Error received from peer ipv4:127.0.0.1:50701 {grpc_message:"invalid ReadRelationshipsRequest.RelationshipFilter: embedded message failed validation | caused by: invalid RelationshipFilter.OptionalSubjectFilter: embedded message failed validation | caused by: invalid SubjectFilter.OptionalSubjectId: value does not match regex pattern \"^(([a-zA-Z0-9/_|\\\\-=+]{1,})|\\\\*)?$\"", grpc_status:3, created_time:"2024-12-13T12:23:08.078194062+00:00"}"
  >
  A 500 error was raised because of <class 'grpc.aio._call.AioRpcError'> on request <Request: DELETE /api/data/groups/z/members/%1D>
```

```
 Reproduce with: 
  E       
  E           curl -X DELETE -H 'host: mockserver:1234' -H 'authorization: ***"is_admin": true, "id": "admin", "name": "Admin Doe", "first_name": "Admin", "last_name": "Doe", "email": "admin.doe@gmail.com", "full_name": "Admin Doe"}' http://localhost/api/data/groups/z/members/%3A
```

--- a/components/renku_data_services/namespace/blueprints.py
+++ b/components/renku_data_services/namespace/blueprints.py
@@ -8,7 +8,7 @@ from sanic_ext import validate
 
 import renku_data_services.base_models as base_models
 from renku_data_services.authz.models import Role, UnsavedMember
-from renku_data_services.base_api.auth import authenticate, only_authenticated
+from renku_data_services.base_api.auth import authenticate, only_authenticated, validate_path_user_id
 from renku_data_services.base_api.blueprint import BlueprintFactoryResponse, CustomBlueprint
 from renku_data_services.base_api.misc import validate_body_root_model, validate_query
 from renku_data_services.base_api.pagination import PaginationRequest, paginate
@@ -148,6 +148,7 @@ class GroupsBP(CustomBlueprint):
         """Remove a specific user from the list of members of a group."""
 
         @authenticate(self.authenticator)
+        @validate_path_user_id
         @only_authenticated
         async def _delete_member(_: Request, user: base_models.APIUser, slug: str, user_id: str) -> HTTPResponse:
             await self.group_repo.delete_group_member(user=user, slug=slug, user_id_to_delete=user_id)

--- a/test/bases/renku_data_services/data_api/test_groups.py
+++ b/test/bases/renku_data_services/data_api/test_groups.py
@@ -271,6 +271,22 @@ async def test_removing_single_group_owner_not_allowed(
 
 
 @pytest.mark.asyncio
+async def test_delete_group_member_invalid(sanic_client: SanicASGITestClient, user_headers: dict[str, str]) -> None:
+    payload = {
+        "name": "demo group",
+        "slug": "demo-group",
+    }
+    _, response = await sanic_client.post("/api/data/groups", headers=user_headers, json=payload)
+    assert response.status_code == 201, response.text
+    group = response.json
+    group_slug = group["slug"]
+
+    _, response = await sanic_client.delete(f"/api/data/groups/{group_slug}/members/%3A", headers=user_headers)
+
+    assert response.status_code == 0, response.text
+
+
+@pytest.mark.asyncio
 async def test_cannot_change_role_for_last_group_owner(
     sanic_client: SanicASGITestClient,
     user_headers: dict[str, str],

--- a/test/bases/renku_data_services/data_api/test_groups.py
+++ b/test/bases/renku_data_services/data_api/test_groups.py
@@ -283,7 +283,7 @@ async def test_delete_group_member_invalid(sanic_client: SanicASGITestClient, us
 
     _, response = await sanic_client.delete(f"/api/data/groups/{group_slug}/members/%3A", headers=user_headers)
 
-    assert response.status_code == 0, response.text
+    assert response.status_code == 422, response.text
 
 
 @pytest.mark.asyncio

